applications/MOF/MOFae.py

```diff
@@ -0,0 +1,287 @@
+import argparse 
+import lbann 
+#import data.mnist 
+import dataset
+import os 
+import lbann.contrib.launcher
+import lbann.contrib.args
+# ----------------------------------
+# Command-line arguments
+# ----------------------------------
+
+## TO DO. Add more CLI arguemnts for Model storage and Generated Data
+
+desc = ("Training 3D-CAE on 4D MOF Data using LBANN")
+
+parser = argparse.ArgumentParser(description = desc)
+
+'''
+parser.add_argument(
+    '--partition', action='store', type=str,
+    help='scheduler partition', metavar='NAME')
+parser.add_argument(
+    '--account', action='store', type=str,
+    help='scheduler account', metavar='NAME')
+'''
+parser.add_argument(
+	'--zdim', action='store',default = 2048, type=int, 
+	help="dimensionality of latent space (dedfault: 2048)", metavar = 'NUM')
+parser.add_argument(
+	'--atoms', action='store', default = 11,type=int, 
+	help="Number of atom species (default: 11)", metavar = 'NUM')
+parser.add_argument(
+    '--job-name', action='store', default='mofae', type=str,
+    help='job name', metavar='NAME')
+parser.add_argument(
+    '--mini-batch-size', action='store', default=128, type=int,
+    help='mini-batch size (default: 128)', metavar='NUM')
+parser.add_argument(
+    '--num-epochs', action='store', default=100, type=int,
+    help='number of epochs (default: 100)', metavar='NUM')
+
+lbann.contrib.args.add_scheduler_arguments(parser)
+'''
+parser.add_argument(
+    '--nodes', action='store', default=4, type=int,
+    help='number of nodes (default: 4)', metavar='NUM')
+parser.add_argument(
+    '--procs_per_node', action='store', default=4, type=int,
+    help='processes per node (default: 4)', metavar='NUM') ##Change to 1 for local machine / openhpc
+parser.add_argument(
+    '--ppn', action='store', default=4,type=int,
+    help='Number of GPUs per node (default: 4)', metavar='NUM') 
+parser.add_argument(
+    '--setup-only' , action='store', default=False, type=bool,
+    help='Setup LBANN experiment without running it')
+parser.add_argument(
+   '--reservation', action='store', default=".", type=str,
+    help='reservation var', metavar='NAME')
+'''
+args = parser.parse_args()
+
+
+latent_dim = args.zdim
+number_of_atoms = args.atoms
+
+
+# ----------------------------------
+# Construct Graph
+# ----------------------------------
+input_ = lbann.Input( target_mode = "reconstruction")
+
+
+tensors = lbann.Identity(input_)
+
+tensors = lbann.Reshape(tensors, dims="11 32 32 32")
+# Input tensor shape is  32x32x32x(number_of_atoms)
+
+# Encoder 
+x = lbann.Convolution(tensors,

```

Comment:
Could you use a for loop here to make it more compact since Conv/BN/Activation are repeated a couple times? You could also simplify it as Encoder/Decoder blocks of Conv/BN/Activation.

Reply:
Yeah definitely. I was following the LeNet convention to explicitly write out layers but that was mostly so I could explicitly see every layer of the model. I can refactor the code.

Reply:
Updated in the last commit to make code more concise